Maestro/src/Microsoft.DotNet.Maestro.WebApi/Controllers/CommitPushedController.cs

```diff
@@ -20,12 +24,22 @@ public class CommitPushedController : ApiController
         private EventRegistrationTable _eventService = new EventRegistrationTable();
         private AzureStorageService _storageService = new AzureStorageService();
 
-        public async Task Post(PushWebHookEvent e)
+        public async Task Post()
         {
+            // The PushWebHookEvent object must be manually deserialized instead of using parameter binding 
+            // because the payload signature verification requires access to the request's raw content.  When
+            // using parameter binding, the content is read during the binding and cannot be accessed again due
+            // to Web API treating the request's content as a forward-only stream which can only be read once.
```

Comment:
I played around with this a bit, and like the aspnet code, I was able to successfully invoke the Request.Content.Read* methods multiple times.  However, when using parameter binding, any attempt to read the content returns empty.  So, it appears that the issue is specific to parameter binding.

I've updated the comment to reflect this finding.

I've also logged https://github.com/aspnet/Mvc/issues/5258 in case it's not the expected ASP.NET behavior.